<?xml version="1.0"?>
<launch>
    <arg name="do_plot" default="False"/>
    <param name="use_sim_time" value="False" type="bool"/>
    <arg name="enable_rviz" default="False"/>

    <arg name="path" default="$(find ekf_rio_tc)/datasets"/>
    <arg name="sequence" default="ColoRadar1"/>

    <arg name="config" default="Coloradar_param"/>
    <arg name="calibration" default="Coloradar_extrinsic"/>

    <arg name="bag_start" default="0"/>
    <arg name="bag_duration" default="-1.0"/>
    <arg name="filter_node_name" default="ekf_rio_tc"/>
    
    <arg name="export_directory" default="$(find ekf_rio_tc)/results"/>
    
    <arg name="log_level" default="Info"/>

    <node name="$(arg filter_node_name)" pkg="ekf_rio_tc" type="rosbag_node" output="screen" launch-prefix="bash -c 'sleep 2; $0 $@' ">
        <rosparam file="$(find ekf_rio_tc)/config/coloradar/$(arg config).yaml" command="load" ns=""/>
        <rosparam file="$(find ekf_rio_tc)/config/coloradar/$(arg calibration).yaml" command="load" ns=""/>
        <param name="rosbag_path" value="$(arg path)/$(arg sequence).bag" type="string"/>
        <param name="bag_start" value="$(arg bag_start)" type="double"/>
        <param name="bag_duration" value="$(arg bag_duration)" type="double"/>
    </node>
    
    <!-- The pose recorder uses the ov_eval tool from the open-vins package -->
    <!--node name="recorder_estimate" pkg="ov_eval" type="pose_to_file" output="screen" required="true">
        <param name="topic"      type="str" value="/$(arg filter_node_name)/pose" />
        <param name="topic_type" type="str" value="PoseStamped" />
        <param name="output"     type="str" value="$(arg export_directory)/$(arg sequence)/stamped_traj_estimate_eval.txt" /-->

    <node pkg="rosservice" type="rosservice" name="set_$(arg filter_node_name)_log_level"
          args="call --wait /$(arg filter_node_name)/set_logger_level 'ros.$(arg filter_node_name)' '$(arg log_level)'"/>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ekf_rio_tc)/config/ekf_rio.rviz" if="$(arg enable_rviz)"/>

</launch>
